### PR TITLE
Tdd

### PR DESCRIPTION
**Pull request by J2ZROMERO**
- [ ] Create a class called Solver.
- [ ] Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- [ ] Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- [ ] Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
- [ ] When N is divisible by 3, return "fizz".
- [ ] When N is divisible by 5, return "buzz".
- [ ] When N is divisible by 3 and 5, return "fizzbuzz".
- [ ] Any other case, return N as a string (e.g. say N is 7 then return "7").
- [ ] TDD approach implemented.
- [ ] Updated unit testing of factorial method to include unit test example when the method parameter is 0.
- [ ] Updated unit testing of factorial method to include unit test example and raise exceptions when the method parameter is negative integer.

### CODE REVIEW

File: spec/factorial_rspec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require '../solver'
+
+describe Solver do
+  context 'When testing the Solver class' do
+    it 'The factorial of 5 should be 120' do
+      solver = Solver.new
+      factorial = solver.factorial(5)
+      expect(factorial).to eq 120
+    end
+  end
```
Comment (sja-thedude) [inline spec/factorial_rspec.rb:10]: - [ ] Kindly also add test for:
The `#factorial` method has tests that handle:
-  Negative value which raises an exception as an expectation
- Special case - when input is 0, the result should equal 1

File: spec/trimmer_decorator_rspec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+require '../person'
+
+describe TrimmerDecorator do
+  describe 'when the trimmer decorator class runs' do
+    it 'returns the correct must give us the first 9 characters of a name' do
+      person = Person.new(10, 'jose juan zepeda Romero')
+      decorator = TrimmerDecorator.new(person)
+      expect(decorator.correct_name).to eq 'jose juan '
+    end
+  end
```
Comment (sja-thedude) [inline spec/trimmer_decorator_rspec.rb:11]: - [ ] Kindly delete the not required files for this project. And only include all the files for TDD.

File: spec/fizzbuzz_rspec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+require '../solver'
```
Comment (sja-thedude) [inline spec/fizzbuzz_rspec.rb:1]: - [ ] Kindly fix this to require './solver' as its not running tests with '../solver'.
Apply the same changes to factorial and reverse as well.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @amare1990, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Correct Gitflow
✔️ Professional Readme

## Required 
- [ ] Kindly remove microverse logo it's not needed.
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/fizzbuzz_rspec.rb (hunk 3 above)
Comment (amare1990) [reply spec/fizzbuzz_rspec.rb:1]: Hi @sja-thedude.
The `solver.rb` file is outside the `spec' folder, i.e., `solver.rb` is in the main directory of our repository. If we did so based on your feedback, it could not work.
We used the command `rspec spec spec/test_file_rspec.rb` in the root directory of our repo., where `test_file_rspec.rb` is the name of the test file.
Thanks

File: spec/trimmer_decorator_rspec.rb (hunk 2 above)
Comment (amare1990) [reply spec/trimmer_decorator_rspec.rb:11]: Hi @sja-thedude . First of all, we would like to thank for reviewing our project milestone and giving us thorough feedback. 
The above file you cited and others are unit testing files of the previous milestone of this project.  We created our branch, TDD on the base of 'dev' branch, where the dev branch currently has codes from the unit-test branch we did and merged yesterday. Therefore, we kindly leave this feedback.
Thanks

**Comment by J2ZROMERO** [comment]
- Removed Microverse logo from Readme file.
- Add negative value which raises an exception as an expectation
- Add Special case - when input is 0, the result should equal 1

**Review by EricMbouwe — APPROVED** [review]
Hi @J2ZROMERO, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Made requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
